hera_cal/tests/test_smooth_cal.py

```diff
@@ -68,6 +69,46 @@ def test_freq_filter(self):
         np.testing.assert_array_equal(ff[0, :], gains[0, :])
         self.assertTrue(info[0]['skipped'])
 
+    def test_time_freq_2D_filter(self):
+        gains = np.ones((10, 10), dtype=complex)
+        gains[3, 5] = 10.0
+        wgts = np.ones((10, 10), dtype=float)
+        wgts[3, 5] = 0
+        freqs = np.linspace(100., 200., 10, endpoint=False) * 1e6
+        times = np.linspace(0, 10 * 10 / 60. / 60. / 24., 10, endpoint=False)
+        ff, info = smooth_cal.time_freq_2D_filter(gains, wgts, freqs, times, filter_mode='rect')
+        np.testing.assert_array_almost_equal(ff, np.ones((10, 10), dtype=complex))
+        ff, info = smooth_cal.time_freq_2D_filter(gains, wgts, freqs, times, filter_mode='plus')
+        np.testing.assert_array_almost_equal(ff, np.ones((10, 10), dtype=complex))
+
+        # test rephasing
+        gains = np.ones((10, 10), dtype=complex)
+        wgts = np.ones((10, 10), dtype=float)
+        gains *= np.exp(2.0j * np.pi * np.outer(150e-9 * np.ones(10), freqs))
+        ff, info = smooth_cal.time_freq_2D_filter(gains, wgts, freqs, times)
+        np.testing.assert_array_almost_equal(ff, gains)
+
+        # test errors
+        with self.assertRaises(ValueError):
+            ff, info = smooth_cal.time_freq_2D_filter(gains, wgts, freqs, times, filter_mode='blah')
+
+    def test_pick_reference_antenna(self):
+        flags = {ant: np.random.randn(10,10)>0 for ant in [(0, 'Jxx'), (1, 'Jxx')]}
```

Comment:
I highly suggest not using completely random numbers in unit tests. An easy solution is to manually set the seed just before, just so it's reproducible if anything goes wrong.

Reply:
Done.